extensions/pkg/webhook/utils_test.go

```diff
@@ -0,0 +1,682 @@
+			var envVars []corev1.EnvVar
+
+			BeforeEach(func() {
+				envVars = []corev1.EnvVar{
+					{
+						Name:  "envVar1",
+						Value: "value1",
+					},
+					{
+						Name:  "envVar2",
+						Value: "value2",
+					},
+				}
+			})
+
+			It("should add a new EnvVar if not present", func() {
+				newEnvVar := corev1.EnvVar{
+					Name:  "envVar3",
+					Value: "value3",
+				}
+				result := webhook.EnsureEnvVarWithName(envVars, newEnvVar)
+				Expect(result).To(Equal(append(envVars, newEnvVar)))
+			})
+
+			It("should replace the existing EnvVar if it's not identical", func() {
+				existingEnvVar := corev1.EnvVar{
+					Name:  "envVar1",
+					Value: "value3",
+				}
+				result := webhook.EnsureEnvVarWithName(envVars, existingEnvVar)
+				Expect(result).To(Equal([]corev1.EnvVar{
+					{
+						Name:  "envVar1",
+						Value: "value3",
+					},
+					{
+						Name:  "envVar2",
+						Value: "value2",
+					},
+				}))
+			})
+
+			It("should do nothing to the existing EnvVar if it's identical", func() {
+				identicalEnvVar := envVars[0]
+				result := webhook.EnsureEnvVarWithName(envVars, identicalEnvVar)
+				Expect(result).To(Equal(envVars))
+			})
+		})
+
+		Describe("EnsureUnitOption", func() {
+			var unitOptions []*unit.UnitOption
+
+			BeforeEach(func() {
+				unitOptions = []*unit.UnitOption{
+					{
+						Section: "Unit",
+						Name:    "Description",
+						Value:   "Test Unit 1",
+					},
+					{
+						Section: "Service",
+						Name:    "ExecStart",
+						Value:   "/usr/bin/test1",
+					},
+				}
+			})
+
+			It("should add a new UnitOption if not present", func() {
+				newOption := &unit.UnitOption{
+					Section: "Install",
+					Name:    "WantedBy",
+					Value:   "multi-user.target",
+				}
+				result := webhook.EnsureUnitOption(unitOptions, newOption)
+				Expect(result).To(Equal(append(unitOptions, newOption)))
+			})
+
+			It("should not replace the existing UnitOption if it's not identical but add it", func() {
+				existingOption := &unit.UnitOption{
+					Section: "Unit",
+					Name:    "Description",
+					Value:   "Test Unit 2",
+				}
+				result := webhook.EnsureUnitOption(unitOptions, existingOption)
+				Expect(result).To(Equal(append(unitOptions, existingOption)))
+			})
+
+			It("should do nothing to the existing UnitOption if it's identical", func() {
+				identicalOption := unitOptions[0]
+				result := webhook.EnsureUnitOption(unitOptions, identicalOption)
+				Expect(result).To(Equal(unitOptions))
+			})
+		})
+
+		Describe("EnsureAnnotationOrLabel", func() {
+			var annotations map[string]string
+
+			BeforeEach(func() {
+				annotations = map[string]string{
+					"annotation1": "value1",
+					"annotation2": "value2",
+				}
+			})
+
+			It("should ensure the specified annotation or label exists", func() {
+				result := webhook.EnsureAnnotationOrLabel(annotations, "annotation3", "value3")
+				Expect(len(result)).To(Equal(3))
+				Expect(result["annotation3"]).To(Equal("value3"))
```

Comment:
```suggestion
				result := webhook.EnsureAnnotationOrLabel(annotations, "annotation3", "value3")
				Expect(result).To(Equal(map[string]string{
					"annotation1": "value1",
					"annotation2": "value2",
					"annotation3": "value3",
				}))
```

or

```suggestion
				Expect(result).To(HaveLen(3))
				Expect(result).To(HaveKeyWithValue("annotation3", "value3"))
```

The same comment applies to the assertions from this Describe block.